### DISCUSSION TITLE
Get Order Metrics Interval Issue

### DISCUSSION THREAD

**Discussion by sktong** [saleweaver/python-amazon-sp-api#407] [Q&A] [answered]
Hi, I use the following code to call the sales end point:
from sp_api.api import Sales
from sp_api.base import Granularity
res = Sales(credentials=credentials).get_order_metrics(
                            interval = "2021-02-20T00:00:00+08:00--2022-02-20T00:00:00+08:00",
                            granularity = Granularity.DAY)
but it comes up with the following errors:

SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': '{getOrderMetrics.arg1=must match "[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{3})?(([+-]\\d\\d:\\d\\d)|Z)(\\[[a-zA-Z/_]+\\])?--[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{3})?(([+-]\\d\\d:\\d\\d)|Z)(\\[[a-zA-Z/_]+\\])?"}', 'details': ''}]

I copy the sample interval format from documentation and I understand the errors may be caused by the bad input of interval but I cannot figure out the correct format. Can anyone help to solve this problem? Many thanks.

**Comment by denisneuf** [comment, marked as answer]
the api takes the interval as a tuple:

def get_order_metrics(self, **interval: tuple**, granularity: Granularity, granularityTimeZone: str = None, **kwargs) -> ApiResponse:

and later will process it as required in your example:

        kwargs.update({
            **'interval': '--'.join([self._create_datetime_stamp(_interval) for _interval in interval]),**
            'granularity': granularity.value,
        })

so your code should be:

```
from sp_api.api import Sales
from sp_api.base import Granularity
res = Sales(credentials=credentials).get_order_metrics(
interval=("2021-02-20T00:00:00+08:00", "2022-02-20T00:00:00+08:00"),
granularity = Granularity.DAY)
```

Let me know if it works for you, in my end it gets the right payload.

**Reply by sktong** [reply]
Thanks so much. It works. I did not realize the interval requires the tuple format.
